### PR TITLE
Documentation update

### LINKED ISSUE #69: C/NOFS IVM list_files
The list_files routine doesn't show up as a class function for ivm data:

cindi = pysat.Instrument("CNOFS", "IVM", clean_level="none")
cindi.list_files()

AttributeError                            Traceback (most recent call last)
<ipython-input-266-089e341e81ff> in <module>()
----> 1 cindi.list_files()

AttributeError: 'Instrument' object has no attribute 'list_files'

### PR DESCRIPTION
**Pull request by jklenzing**
Closes #69, #94, and #153.

Summary of changes
- Added instructions to list files in new_instrument.rst (#69)
- Added description of sat_ids to new_instrument.rst (#94)
- Scrubbed code in demo files (pep8, python2 -> 3)
- Updated code in tutorial.rst and examples.rst (#153), making examples consistent with demos
- Updated images for examples.rst, using Perceptually Uniform Sequential colormaps
- Updated discussion of commands in documentation taking into account the utils reorganization in #164

### CODE REVIEW

**Comment by jklenzing** [comment]
Direct links to changed documents:
https://github.com/rstoneback/pysat/blob/documentation/docs/examples.rst
https://github.com/rstoneback/pysat/blob/documentation/docs/new_instrument.rst
https://github.com/rstoneback/pysat/blob/documentation/docs/tutorial.rst

File: docs/examples.rst [hunk 1]
```diff
@@ -10,78 +10,85 @@ How often does a particular thing occur on a orbit-by-orbit basis? Let's find ou
...
-   start = pds.datetime(2010,5,9)
-   stop = pds.datetime(2010,5,15)
-
-   # if there is no vefi dc magnetometer data on your system 
-   # run command below
-   # where start and stop are pandas datetimes (from above)
-   # pysat will automatically register the addition of this data at the end    
-   # of download
-   vefi.download(start, stop)
+  '''
```
Comment (aburrell) [inline docs/examples.rst]: ''' should be """

Comment (jklenzing) [reply docs/examples.rst]: fixed here and in demos

**Comment by jklenzing** [comment]
> Other comments relate to content of new_instrument.rst:
> 
>     * Supported Data Templates should include Madrigal
> 


Added.  Madrigal currently does not use `functools.partial`.  Is this something we should update?

Writing up documentation for current methods.

**Comment by jklenzing** [comment]
Updates completed.  We should have a larger conversation about how the acknowledgements are implemented.

File: docs/new_instrument.rst [hunk 2]
```diff
@@ -26,18 +33,26 @@ Pysat maintains a list of files to enable data management functionality. It need
...
 
 .. code:: python
 
    def list_files(tag=None, data_path=None):
-       return pysat.Files.from_os(data_path=data_path, 
+       return pysat.Files.from_os(data_path=data_path,
                     format_str='cindi-{year:4d}{day:03d}-ivm.hdf')
-                                
+
+The list_files function can be invoked from within ipython by calling
```
Comment (rstoneback) [inline docs/new_instrument.rst]: Technically the information in instrument.files.files comes from list_files, but you aren't directly invoking the method.

If a user wants pysat to run list_files when the Instrument is instantiated, then pass update_files=True.

File: docs/new_instrument.rst [hunk 3]
```diff
@@ -179,15 +214,52 @@ Using the CDAWeb template the equivalent action is
...
+   download = functools.partial(nasa_cdaweb_methods.download,
                                 supported_tags)
 
 where supported_tags is defined as dictated by the download function. See the routines for cnofs_vefi and cnofs_ivm for practical uses of the NASA CDAWeb support code.
 |br|
 
 
 .. automodule:: pysat.instruments.nasa_cdaweb_methods
-   :members: 
+   :members:
```
Comment (rstoneback) [inline docs/new_instrument.rst]: There is a formatting issue that starts here. The stuff below is rendered as is, not as formatted text.

Comment (jklenzing) [reply docs/new_instrument.rst]: fixed

File: docs/new_instrument.rst (hunk 2 above)
Comment (jklenzing) [reply docs/new_instrument.rst]: added some text about this

**Comment by jklenzing** [comment]
Updates to madrigal are out of scope here.  Adding a note to #106.

**Pull request merged**